packages/react-components/src/components/table-output-component.tsx

```diff
@@ -653,4 +654,116 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
             });
         }
     }
+
+    async shareOutput(option: string): Promise<void> {
+        if (!this.gridApi) {
+            return;
+        }
+
+        const traceUUID = this.props.traceId;
+        const tspClient = this.props.tspClient;
+        const outputId = this.props.outputDescriptor.id;
+        let fetchIndex = 0;
+        let totalLinesToFetch = this.gridApi.getDisplayedRowCount();
+        const linesArray = new Array<Line>();
+        const bufferSize = 100000;
+
+        // Convert to csv
+        const csv_data = [];
+
+        // Fetch column headers
+        const colHeaderRow: string[] = [];
+        this.columnArray.forEach(columnHeader => {
+            colHeaderRow.push(columnHeader.headerName);
+        });
+        csv_data.push(colHeaderRow.join(','));
+
+        if (option === 'selection') {
+            if (this.startTimestamp && this.endTimestamp) {
+                const startIndex = await this.fetchTableIndex(this.startTimestamp < this.endTimestamp ? this.startTimestamp : this.endTimestamp);
+                const endIndex = await this.fetchTableIndex(this.startTimestamp > this.endTimestamp ? this.startTimestamp : this.endTimestamp);
```

Comment:
Not working properly. The export should be index-based not time-based. The user could have made a specific index selection and when multiple events share the same timestamp this does not export the correct rows.

Reply:
@PatrickTasse Updated PR to support index-based selections as well

Reply:
Should it not be index-based immediately? Not sure if you are going to merge commits of this PR?